paderbox/io/new_subdir.py

```
@@ -0,0 +1,341 @@
+        consider_mpi:
+            If True, only search on one mpi process for the folder and
+            distribute the folder name.
+            When using mpi with this option disables the following can/will
+            happen
+             - When mkdir is True every process will get another folder.
+               i.e. each process has a folder just for this process.
+             - When mkdir is False it is likely that all get the same folder,
+               but this is not guarantied. Never use mpi with consider_mpi and
+               mkdir disabled.
+        dry_run:
+            Similar to mkdir. When true, disable mkdir and print the folder
+            name.
+
+    Returns:
+        pathlib.Path of the new subdir
+
+
+    >>> get_new_subdir('/', dry_run=True)  # root folder usually contain no digits
+    dry_run: "os.mkdir(/1)"
+    PosixPath('/1')
+
+    >>> import numpy as np
+    >>> np.random.seed(0)  # This is for doctest. Never use it in practise.
+    >>> get_new_subdir('/', id_naming=NameGenerator(), dry_run=True)
+    dry_run: "os.mkdir(/helpful_tomato_finch)"
+    PosixPath('/helpful_tomato_finch')
+    """
+
+    if consider_mpi:
+        import dlp_mpi
+        if dlp_mpi.IS_MASTER:
+            pass
+        else:
+            new_folder = None
+            new_folder = dlp_mpi.bcast(new_folder)
+            return new_folder
+
+    basedir = Path(basedir).expanduser().resolve()
+    if not basedir.exists():
+        if dry_run:
+            print(f'dry_run: "os.makedirs({basedir})"')
+            # ToDo: Make this working.
+            #       Will fail when calling os.listdir
+        else:
+            basedir.mkdir(parents=True)
+
+    if Path('/net') in basedir.parents:
+        # If nt filesystem, assert not in /net/home
+        assert Path('/net/home') not in basedir.parents, basedir
+
+    prefix_ = f'{prefix}_' if prefix else ''
+    _suffix = f'_{suffix}' if suffix else ''
+
+    for i in range(200):
```

The 200 are a bit arbitrary, should we set a variable somewhere so that it can be changed more easily?

---

It is even more arbitrary, because later the code ends the loop with the check `i > 100`.
I don't think we have to expose this parameter.

When someone gets a problem with this number, we can change it.
From my observations, it fails fast enough, when something is wrong and launching 100 experiments at the same time is difficult (When you launch with a delay of 10 ms it is unlikely that you get more than 1 collision).

---

Just for documentation: The origin for this loop is https://github.com/IDSIA/sacred/pull/285.